oauth2/oauth2-jose/src/test/java/org/springframework/security/oauth2/jwt/NimbusJwtDecoderTests.java

```diff
@@ -559,6 +560,22 @@ public void decodeWhenUsingSecretKeyWithKidThenStillUsesKey() throws Exception {
 		// @formatter:on
 	}
 
+	// gh-7056
+	@Test
+	public void decodeWhenUsingJwkSource() throws Exception {
```

Comment:
To follow the naming convention in this file, will you please follow this format for the method name:

```java
{methodName}When{Condition}Then{Expectation}
```

For example:

```java
withJwkSourceWhenDefaultsThenUsesProvidedJwkSource
```